src/core/cache.jl

```diff
@@ -11,18 +14,51 @@ function TimeStatusChange(parameter::Symbol)
     return TimeStatusChange(value_array, UpdateRef{PJ.ParameterRef}(parameter))
 end
 
-cache_value(cache::AbstractCache, key) = cache.value[key]
+function TimeStatusChange(ref::UpdateRef)
+    value_array = JuMP.Containers.DenseAxisArray{Dict{Symbol, Float64}}(undef, 1)
+    return TimeStatusChange(value_array, ref)
+end
+
+mutable struct DeviceCommitment <: AbstractCache
+    value::JuMP.Containers.DenseAxisArray{Float64}
+    ref::UpdateRef
+end
+
+function DeviceCommitment(ref::UpdateRef)
+    value_array = JuMP.Containers.DenseAxisArray{Float64}(undef, 1)
+    return DeviceCommitment(value_array, ref)
+end
+
+mutable struct DeviceLevel <: AbstractCache
+    value::JuMP.Containers.DenseAxisArray{Float64}
+    ref::UpdateRef
+end
+
+function DeviceLevel(ref::UpdateRef)
+    value_array = JuMP.Containers.DenseAxisArray{Float64}(undef, 1)
+    return DeviceLevel(value_array, ref)
+end
+
+CacheKey(cache::TimeStatusChange) = CacheKey(InitialConditionCache, cache.ref)
+CacheKey(cache::AbstractCache) = CacheKey(FeedForwardCache, cache.ref)
+
+cache_value(cache::AbstractCache, key...) = cache.value[key...]
+
+function build_cache!(cache::TimeStatusChange, sim::Simulation, stage_name::String)
+    build_cache!(cache, get_psi_container(get_stage(sim,stage_name)))
+end
 
 function build_cache!(cache::TimeStatusChange, op_problem::OperationsProblem)
     build_cache!(cache, op_problem.psi_container)
 end
 
 function build_cache!(cache::TimeStatusChange, psi_container::PSIContainer)
     parameter = get_value(psi_container, cache.ref)
-    value_array = JuMP.Containers.DenseAxisArray{Dict{Symbol, Float64}}(undef, axes(parameter)...)
+    value_array = JuMP.Containers.DenseAxisArray{Dict{Symbol, Float64}}(undef, parameter.axes[1])
```

Comment:
Why is this hard-coded to 1?

Reply:
Current Implementation will lead to parameter's having 2 axes but only the 1st (Device names) is required in creating this type of  Cache

Reply:
This needs to be reverted to the previous code.